src/components/input/template.test.js

```diff
@@ -0,0 +1,113 @@
+/* globals describe, it, expect */
+
+const { render, getExamples } = require('../../../lib/jest-helpers')
+
+const examples = getExamples('input')
+
+describe('Input', () => {
+  describe('by default', () => {
+    it('renders with classes', () => {
+      const { $ } = render('input', {
+        classes: 'app-c-input--custom-modifier'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.hasClass('app-c-input--custom-modifier')).toBeTruthy()
+    })
+
+    it('renders with id', () => {
+      const { $ } = render('input', {
+        id: 'my-input'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('id')).toEqual('my-input')
+    })
+
+    it('renders with name', () => {
+      const { $ } = render('input', {
+        name: 'my-input-name'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('name')).toEqual('my-input-name')
+    })
+
+    it('renders with type="text"', () => {
+      const { $ } = render('input')
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('type')).toEqual('text')
+    })
+
+    it('renders with value', () => {
+      const { $ } = render('input', {
+        value: 'QQ 12 34 56 C'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.val()).toEqual('QQ 12 34 56 C')
+    })
+
+    it('renders with attributes', () => {
+      const { $ } = render('input', {
+        attributes: {
+          'aria-describedby': 'my-label'
+        }
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('aria-describedby')).toEqual('my-label')
+    })
+
+    it('renders with label', () => {
+      const { $ } = render('input', {
+        id: 'my-input',
+        label: {
+          'text': 'National Insurance number'
+        }
+      })
+
+      const $label = $('.govuk-c-label')
+      expect($label).toBeTruthy()
+      expect($label.attr('for')).toEqual('my-input')
```

Comment:
Again, suggest this is split into two parts - one that tests that the label is rendered correctly (probably worth checking the text itself rather than just ensuring the element exists?) and a second test that ensures the label is correctly associated with the input?

Reply:
I expect the label to be tested within the label macro. We don't add any logic about the label in the input macro, except for setting label's `for` attribute equal to the input's `id`.

Reply:
OK. I still think it's worth having an explicit test for that for/id relationship.

Reply:
split